solidity/core/contracts/validator-manager/MultisigValidatorManager.sol

```diff
@@ -0,0 +1,239 @@
+     * the checkpoint.
+     */
+    function isQuorum(
+        bytes32 _root,
+        uint256 _index,
+        bytes[] calldata _signatures
+    ) public view returns (bool) {
+        uint256 _signaturesLength = _signatures.length;
+        // If there are less signatures provided than the required quorum threshold,
+        // this is not a quorum.
+        if (_signaturesLength < quorumThreshold) {
+            return false;
+        }
+        // To identify duplicates, the signers recovered from _signatures
+        // must be sorted in ascending order. previousSigner is used to
+        // enforce ordering.
+        address _previousSigner = address(0);
+        uint256 _validatorSignatureCount = 0;
+        for (uint256 i = 0; i < _signaturesLength; i++) {
+            address _signer = _recoverCheckpointSigner(
+                _root,
+                _index,
+                _signatures[i]
+            );
+            // Revert if the signer violates the required sort order.
+            require(_previousSigner < _signer, "!sorted signers");
+            // If the signer is a validator, increment _validatorSignatureCount.
+            if (validators.contains(_signer)) {
+                _validatorSignatureCount++;
+            }
+            _previousSigner = _signer;
+        }
+        return _validatorSignatureCount >= quorumThreshold;
+    }
+
+    // ============ Internal Functions ============
+
+    /**
+     * @notice Recovers the signer from a signature of a checkpoint.
+     * @param _root The checkpoint's merkle root.
+     * @param _index The checkpoint's index.
+     * @param _signature Signature on the the checkpoint.
+     * @return The signer of the checkpoint signature.
+     **/
+    function _recoverCheckpointSigner(
+        bytes32 _root,
+        uint256 _index,
+        bytes calldata _signature
+    ) internal view returns (address) {
+        bytes32 _digest = keccak256(
+            abi.encodePacked(outboxDomainHash, _root, _index)
+        );
+        _digest = ECDSA.toEthSignedMessageHash(_digest);
+        return ECDSA.recover(_digest, _signature);
+    }
+
+    /**
+     * @notice Enrolls a validator into the validator set.
+     * @dev Reverts if _validator is already in the validator set.
+     * @param _validator The validator to add to the validator set.
+     */
+    function _enrollValidator(address _validator) internal {
+        require(validators.add(_validator), "already enrolled");
+        emit EnrollValidator(_validator);
+    }
+
+    /**
+     * @notice Unenrolls a validator from the validator set.
+     * @dev Reverts if the resulting validator set length is less than
+     * the quorum threshold.
+     * @dev Reverts if _validator is not in the validator set.
+     * @param _validator The validator to remove from the validator set.
+     */
+    function _unenrollValidator(address _validator) internal {
+        require(
```

Comment:
Yeah I initially thought something along the same lines. My main reasons are:

1. Imagine a controller/governor owns all the validator managers on chain A and chain B. The controller/governor chain is chain A. If you're unenrolling a validator for the InboxValidatorManager on chain B that tracks the chain A validator set, you could accidentally violate the quorum threshold, which would then brick all future controller messages from chain A to chain B. Now the only way to lower the quorum threshold to regain liveness is to activate the chain B recovery manager :/
2. Because the controller/governor allows for multiple calls to be executed atomically, I think it's reasonable and safe enough to require `call[0]` to lower the quorum threshold and `call[1]` to unenroll the validator